### PR TITLE
check param, result, type param of function in redefines-builtin-id rule

### LINKED ISSUE #1022: Request: report "redefinition of the built-in" errors for function related variables
**Is your feature request related to a problem? Please describe.**

Currently, function related variables are not checked by `redefinition of the built-in ...`.
It would be nice if revive reports `redefinition of the built-in ...` errors for function related variables.

For example, revive does not report any errors for the following `main.go` file:
```golang
// This is a package comment
package main

func main() {
}

// 1 param
func foo1(new int) {
	_ = new
}

// 2 result
func foo2() (new int) {
	return
}

// 3 type param
func foo3[new any]() {
}
```

```
❯ revive -set_exit_status main.go

❯ echo $?
0
```

**Describe the solution you'd like**

revive reports `redefinition of the built-in ...` errors for function related variables, so that:

```
❯ revive -set_exit_status main.go
main.go:8:11: redefinition of the built-in function new
main.go:13:14: redefinition of the built-in function new
main.go:18:11: redefinition of the built-in function new

❯ echo $?
1
```

**Describe alternatives you've considered**

NA

**Additional context**

I'm not completely certain, but I believe that by making the following changes, this feature can be implemented.

```diff
diff --git rule/redefines-builtin-id.go rule/redefines-builtin-id.go
index b3ff084..d0870fd 100644
--- rule/redefines-builtin-id.go
+++ rule/redefines-builtin-id.go
@@ -125,6 +125,29 @@ func (w *lintRedefinesBuiltinID) Visit(node ast.Node) ast.Visitor {
 		if ok, bt := w.isBuiltIn(id); ok {
 			w.addFailure(n, fmt.Sprintf("redefinition of the built-in %s %s", bt, id))
 		}
+	case *ast.FuncType:
+		var fields []*ast.Field
+		if n.TypeParams != nil {
+			fields = append(fields, n.TypeParams.List...)
+		}
+		if n.Params != nil {
+			fields = append(fields, n.Params.List...)
+		}
+		if n.Results != nil {
+			fields = append(fields, n.Results.List...)
+		}
+		for _, field := range fields {
+			for _, name := range field.Names {
+				if obj := name.Obj; obj != nil {
+					if obj.Kind == ast.Var || obj.Kind == ast.Typ {
+						id := obj.Name
+						if ok, bt := w.isBuiltIn(id); ok {
+							w.addFailure(name, fmt.Sprintf("redefinition of the built-in %s %s", bt, id))
+						}
+					}
+				}
+			}
+		}
 	case *ast.AssignStmt:
 		for _, e := range n.Lhs {
 			id, ok := e.(*ast.Ident)
```

### PR DESCRIPTION
**Pull request by skaji**
Close #1022 

Currently, function related variables (param, result, and type param) are not checked by `redefines-builtin-id rule`.
It would be nice if revive reports `redefinition of the built-in ...` errors for function related variables.

### CODE REVIEW

File: rule/redefines-builtin-id.go [hunk 1]
```diff
@@ -125,6 +125,29 @@ func (w *lintRedefinesBuiltinID) Visit(node ast.Node) ast.Visitor {
...
+		if n.Params != nil {
+			fields = append(fields, n.Params.List...)
+		}
+		if n.Results != nil {
+			fields = append(fields, n.Results.List...)
+		}
+		for _, field := range fields {
+			for _, name := range field.Names {
+				if obj := name.Obj; obj != nil {
+					if obj.Kind == ast.Var || obj.Kind == ast.Typ {
```
Comment (denisvmedia) [inline rule/redefines-builtin-id.go:142]: nit: these conditions can be put in the previous if.

Comment (skaji) [reply rule/redefines-builtin-id.go]: @denisvmedia Thank you for the review.
I combined the if statements in 7a0e47f.

**Review by denisvmedia — APPROVED** [review]
LGTM, will give a chance for at least one other review

**Comment by skaji** [comment]
I merged "master" branch to resolve conflict.

**Pull request merged**